pkg/tfbridge/schema.go

```diff
@@ -494,8 +493,6 @@ func (ctx *conversionContext) makeTerraformInput(
 
 		var tfflds shim.SchemaMap
 
-		// We cannot use [shimutil.CastToTypeObject] because we have machinery that constructs invalid
```

Comment:
This is exactly the root cause, I find it unfortunate to remove this comment. `elemSchemas` produces

```go
(&schema.Schema{Elem: e}).Shim()
```

This is not a valid schema as it does not have `Type()` defined. This is an artificial schema that you will not find coming from the actual provider, it's a recursion artifact.

Reply:
But CastToTypeObject is wrong in the first place for SDKv2 - it checks for .Resource under a Map type which is illegal in SDKV2.

I don't understand why we'd want to use it here anyway - `tfs.Elem().(shim.Resource)` is the correct way to check for an object type

Reply:
I'm not going to block on this change but the Elem() representation is continuously confusing.  I don't think "object type" is defined enough for me. 

How about https://github.com/pulumi/pulumi-terraform-bridge/pull/2187 to try to clarify some of the issues here?

Reply:
My understanding was that `tfs.Elem().(shim.Resource)` was insufficient to identify an object type: `tfs.Type() == TypeList && tfs.Elem().(shim.Resource)` implied a `List[Object]` where `tfs.Type() == TypeMap && tfs.Elem().(shim.Resource)` implied an `Object` type.

Reply:
When recurring over a List[Object] type which is indeed `tfs.Type() == TypeList && tfs.Elem().(shim.Resource)` the code will call elemSchemas to compute a shim.Schema for the Object, which will result in a shim.Schema value that's not recognized by CastToTypeObject. One can argue that it should, in fact, be recognized, and then we could bring that back here and work this out that way. That's a bit more changes though.